### PR TITLE
modified tests to run scheduling on platforms not supporting execution

### PR DESCRIPTION
**Pull request by gilbo**
This will eliminate upload conflicts for me running on my mac without access to AVX2, AVX512, or AMX.  I can at least catch scheduling errors this way and fix the tests without triggering failed CI tests.

### CODE REVIEW

File: tests/amx/test_amx_instr.py [hunk 1]
```diff
@@ -348,6 +349,12 @@ def test_matmul_on_amx_scheduled_i8(compiler, sde64):
     print("Final scheduled algorithm:")
     print(amx)
 
+    return amx
+
+def test_gen_matmul_i8_amx(matmul_i8):
```
Comment (yamaguchi1024) [inline tests/amx/test_amx_instr.py:354]: @gilbo Why do you need this?

Comment (gilbo) [reply tests/amx/test_amx_instr.py:354]: This ensures that the scheduling code will get executed on my machine even though I don't have AMX installed.  Then I can catch errors arising from the scheduling code iteslf, even if I can't run a compilation check to make sure the result is still correct.

Comment (yamaguchi1024) [reply tests/amx/test_amx_instr.py:354]: Ahh so it's like a wrapper around even though you don't have `compiler`

**Review by yamaguchi1024 — APPROVED** [review]
Thanks!

**Pull request merged**